bindings/python/tests/test_parse.py

```diff
@@ -32,3 +32,179 @@ def get_parsed_info(tmp_path, file_contents):
+
+    assert struct_decl["kind"] == "STRUCT_DECL"
+    assert struct_decl["name"] == "AStruct"
+
+
+def test_cxx_base_specifier(tmp_path):
+    file_contents = """
+    struct BaseStruct {};
+    struct DerivedStruct: public BaseStruct {};
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    child_struct_decl = parsed_info["members"][1]
+    cxx_base_specifier = child_struct_decl["members"][0]
+
+    assert cxx_base_specifier["kind"] == "CXX_BASE_SPECIFIER"
+    assert cxx_base_specifier["access_specifier"] == "PUBLIC"
+    assert cxx_base_specifier["name"] == "struct BaseStruct"
+
+
+def test_cxx_method(tmp_path):
+    file_contents = """
+    struct AStruct {
+        void aMethod() {}
+    };
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    struct_decl = parsed_info["members"][0]
+    cxx_method = struct_decl["members"][0]
+
+    assert cxx_method["kind"] == "CXX_METHOD"
+    assert cxx_method["result_type"] == "void"
+    assert cxx_method["name"] == "aMethod"
+
+
+def test_type_ref(tmp_path):
+    file_contents = """
+    struct AStruct { 
+        int aClassMember;
+        void aMethod(AStruct& aClassMember) {}
+    };
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    struct_decl = parsed_info["members"][0]
+    cxx_method = struct_decl["members"][1]
+    l_value_ref = cxx_method["members"][0]
+    type_ref = l_value_ref["members"][0]
+
+    assert type_ref["kind"] == "TYPE_REF"
+    assert type_ref["name"] == "struct AStruct"
+
+
+def test_constructor(tmp_path):
+    file_contents = """
+    struct AStruct {
+        AStruct() {}
+    };
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    struct_decl = parsed_info["members"][0]
+    constructor = struct_decl["members"][0]
+
+    assert constructor["kind"] == "CONSTRUCTOR"
+    assert constructor["access_specifier"] == "PUBLIC"
+    assert constructor["name"] == "AStruct"
+
+
+def test_parm_decl(tmp_path):
+    file_contents = """
+    struct AStruct {
+        AStruct(int aFunctionArgument) {}
+    };
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    struct_decl = parsed_info["members"][0]
+    constructor = struct_decl["members"][0]
+    parm_decl = constructor["members"][0]
+
+    assert parm_decl["kind"] == "PARM_DECL"
+    assert parm_decl["element_type"] == "Int"
+    assert parm_decl["name"] == "aFunctionArgument"
+
+
+def test_unexposed_expr(tmp_path):
+    file_contents = """
+    struct AStruct {
+        int aClassMember;
+        AStruct(int aFunctionArgument) : aClassMember(aFunctionArgument) {}
+    };
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    struct_decl = parsed_info["members"][0]
+    constructor = struct_decl["members"][1]
+    unexposed_expr = constructor["members"][2]
```

Comment:
What is `constructor[members][1]`? Previous test case shows what 0 is. This one shows what 2 is.

Reply:
`member_ref`:
```
{'depth': 3, 'line': 4, 'column': 65, 'kind': 'MEMBER_REF', 'name': 'aClassMember', 'element_type': 'Int', 'members': []}
```

Used a different source for that:
```py
def test_member_ref(tmp_path):
    file_contents = """
    struct AStruct {
        int firstMember, secondMember;
        AStruct(int firstFunctionParameter, int secondFunctionParameter)
        : firstMember(secondFunctionParameter), secondMember(firstFunctionParameter)
        {}
    };
    """
    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
    struct_decl = parsed_info["members"][0]
    constructor = struct_decl["members"][2]
    member_ref_1 = constructor["members"][2]
    member_ref_2 = constructor["members"][4]

    assert member_ref_1["kind"] == "MEMBER_REF"
    assert member_ref_2["kind"] == "MEMBER_REF"
    assert member_ref_1["element_type"] == "Int"
    assert member_ref_2["element_type"] == "Int"
    assert member_ref_1["name"] == "firstMember"
    assert member_ref_2["name"] == "secondMember"
```

Reply:
This test also misses on what the skipped element (`constructor["members"][odd_number]`) is

Reply:
I should have been more clear.
The skipped element is:
> `{'depth': 3, 'line': 4, 'column': 65, 'kind': 'MEMBER_REF', 'name': 'aClassMember', 'element_type': 'Int', 'members': []}
`

I tested this initially in the `test_member_ref` function, but later on, changed the example in the function to a more detailed one.

Old test:https://github.com/divmadan/pcl/blob/74b266024aaf69969660c7d1fa1fd3ef7ddceef3/bindings/python/tests/test_parse.py#L260-L275

Reply:
Got it. May I recommend adding at least the name for all member_ref if not the detailed checks